### PR TITLE
Refactor documentation

### LINKED ISSUE #76: Update documentation for installing perccli
Currently doc suggests to download perccli from [here]( https://www.dell.com/support/home/zh-tw/drivers/driversdetails?driverid=52r3d) and then convert the rpm to deb using alien.

We should update the doc to suggest to download the tool from [this page](https://www.dell.com/support/home/en-uk/drivers/driversdetails?driverid=36g6n) where the debian package is already present and then we can drop instructions to convert the rpm.

### PR DESCRIPTION
**Pull request by dashmage**
Resolves #76

### CODE REVIEW

**Comment by dashmage** [comment]
Note: There's no updating to be done on the discourse post. See similar PR #56 for more reference.

**Review by chanchiwai-ray — APPROVED** [review]
LGTM

**Review by Pjack — COMMENTED** [review]
Should we maintain this information in two place. (charmhub and source code).
Either 
1. we could attach charmhub link into this one.
2. Just remove it. We maintain user manual in charmhub. In source code, we just leave the minimum document for developer.
* What is it
* How to build it
* Design of architecture
* User manual ->  link to charmhub
What do you think?

**Review by Pjack — CHANGES_REQUESTED** [review]
Should we maintain this information in two places. (charmhub and source code).
We maintain user manual in charmhub. So I prefer to attach charmhub link is good enough.
In source code, we maintain the document for developer.

What is it
How to build it
Design of architecture
User manual -> link to charmhub
What do you think?

**Comment by chanchiwai-ray** [comment]
Thanks for pointing it out. I did not notice that we have a `doc` directory. What is the `doc` directory, and why we are putting the download link there? I thought we should put the download link in the discourse post somewhere (e.g. `Tutorial` tab) telling the end-user how to properly upload the resource.

> Should we maintain this information in two places. (charmhub and source code). We maintain user manual in charmhub. So I prefer to attach charmhub link is good enough. In source code, we maintain the document for developer.
> 
> What is it How to build it Design of architecture User manual -> link to charmhub What do you think?
> 
> note: I hope to finish this discussion before we merge it, so I change the status to `Changes requested`. But there is nothing have to be changed here.

I thought this is our current practice.

- Discourse post for end-user 
  - Tutorial
  - Metrics
  - ...
- `readme.md` for developer
  - What is it
  - How to build it
  - Design of architecture

**Comment by jneo8** [comment]
> Thanks for pointing it out. I did not notice that we have a doc directory. What is the doc directory, and why we are putting the download link there? I thought we should put the download link in the discourse post somewhere (e.g. Tutorial tab) telling the end-user how to properly upload the resource.
>
>
> Should we maintain this information in two places. (charmhub and source code).
We maintain user manual in charmhub. So I prefer to attach charmhub link is good enough.


I have different opinion here. The documents should always follow the newest repository. So there should always have the newest documents in the repository and we should review the document change in the PR. Most of the projects have this practice.

The discourse.charmhub for me is an extra document for end user. I prefer to maintain two documents but the document on github is the first priority.

**Comment by chanchiwai-ray** [comment]
After double checking the "Charm development best practices". I think I might misunderstand the purpose of `readme.md` for a charm.

> The discourse.charmhub for me is an extra document for end user. 

@jneo8 👍🏼 , this is described in the best practice.

[1] https://juju.is/docs/sdk/styleguide#heading--documentation
[2] https://juju.is/docs/sdk/charm-documentation

@Pjack Are we following these best practice?

**Comment by dashmage** [comment]
> Should we maintain this information in two places. (charmhub and source code)？ We maintain user manual in charmhub. So I prefer to attach charmhub link is good enough. In source code, we maintain the document for developer.
> 
> What is it How to build it Design of architecture User manual -> link to charmhub What do you think?
> 
> note: I hope to finish this discussion before we merge it, so I change the status to `Changes requested`. But there is nothing have to be changed here.

Here are my thoughts:

- We should prioritise maintaining up-to-date docs in one place. It will be hard to propagate changes to multiple locations when needed. I believe we should prioritise the charmhub page (via the contents of the discourse post). The README can contain all the basic info and we can link to the charmhub page for the user to get more details. 
- As per our current best practices, we're following the [Diátaxis](https://diataxis.fr/) approach to technical documentation writing in discourse. This approach is supposed to be comprehensive and for anyone to consume from end-user to developer. 
- The 4 sections (Tutorial, How-To, Explanation, Reference) will have all the details any type of user will need. The end-user using the application can go through the "Tutorial" (eg: getting started with basic deployment) or "Explanation" (eg: building the app) sections whereas if any developer wants to dig deeper, they can consult the "How-To" Guides (eg: how to download and upload resources) and "Reference" sections (eg: architectural design of app)

**Comment by jneo8** [comment]
> > Should we maintain this information in two places. (charmhub and source code)？ We maintain user manual in charmhub. So I prefer to attach charmhub link is good enough. In source code, we maintain the document for developer.
> > What is it How to build it Design of architecture User manual -> link to charmhub What do you think?
> > note: I hope to finish this discussion before we merge it, so I change the status to `Changes requested`. But there is nothing have to be changed here.
> 
> Here are my thoughts:
> 
> * We should prioritise maintaining up-to-date docs in one place. It will be hard to propagate changes to multiple locations when needed. I believe we should prioritise the charmhub page (via the contents of the discourse post). The README can contain all the basic info and we can link to the charmhub page for the user to get more details.
> * As per our current best practices, we're following the [Diátaxis](https://diataxis.fr/) approach to technical documentation writing in discourse. This approach is supposed to be comprehensive and for anyone to consume from end-user to developer.
> * The 4 sections (Tutorial, How-To, Explanation, Reference) will have all the details any type of user will need. The end-user using the application can go through the "Tutorial" section (eg: getting started with basic deployment) or "How-To" Guides (eg: how to download and upload resources) whereas if any developer wants to dig deeper, they can consult the "Explanation" (eg: building the app) and "Reference" sections (eg: architectural design of app)
> 
> Edit: After checking the links shared by @chanchiwai-ray , I think we should
> 
> * Use README as a minimal starting point which is the first thing user might look at. Basic setup, related charms to be deployed, common configs etc.
> * Link to the discourse page in the README for the user to get more detailed info.

- The first priority should be repository on github.
- The `README.md` can be what you described([diataxis](https://diataxis.fr/) is nice). But other documents should also been put inside the repository as possible from my point of view. At least I don't find any disadvantage to do this. Charmhub's document somehow give more maintain efforts during the version updating.

File: docs/resources.md [hunk 1]
```diff
@@ -1,12 +1,7 @@
 # PercCLI
 
-- Download from: https://www.dell.com/support/home/zh-tw/drivers/driversdetails?driverid=52r3d
-- unzip
-
-- Use `alien` to transform `rpm` to `.deb`.
-    ```sh
-    sudo alien -c ./perccli-007.0318.0000.0000-1.noarch.rpm
-    ```
+- Download from: https://www.dell.com/support/home/en-us/drivers/driversdetails?driverid=36g6n
```
Comment (jneo8) [inline docs/resources.md:3]: **suggestion (blocking):** 

I found newer versions on the website:

- [7.2313.0000.0000, A14, released at 2023/05/18](  https://www.dell.com/support/home/zh-tw/drivers/driversdetails?driverid=tdghn)
- [7.2110.0.0, A13, release at 2022/11/14](https://www.dell.com/support/home/zh-tw/drivers/driversdetails?driverid=mgkfv)

We should provide the newest link.

**Review by jneo8 — CHANGES_REQUESTED** [review]
I found two newest versions for PERCCLI, the one

File: docs/resources.md (hunk 1 above)
Comment (dashmage) [reply docs/resources.md:3]: Good suggestion. I've confirmed that the deb package is also present in the newer `A14` version. I'll update the documentation with that version

**Comment by Pjack** [comment]
Thanks for the comment and opinion from everyone. I understand the pros and cons of each solution. 
Based on the discussion in the team and follow the recommandation in [charm-documentation](https://juju.is/docs/sdk/charm-documentation).

* A quick getting started guide for a simple deployment
* Mandatory configuration steps for minimal deployment - such as required relationships
* Links to more detailed documentation (like a [Discourse](https://juju.is/docs/sdk/charm-documentation#heading--detailed-docs) post - see below)

Therefore, let's describe the basic steps in readme.md and then link to charmhub(discourse). 

Furthermore, Just like Ray mentioned, I misunderstand the purpose of `readme.md`. We should put deverlop-specific instructions in another document, e.g. CONTRIBUTING.md

https://juju.is/docs/sdk/styleguide#heading--documentation
Put this information in other documentation pages specific to the development of and contribution to your charm. This information can live as part of your [Charm Documentation](https://juju.is/docs/sdk/charm-documentation), or in the version control repository for your charm ([example](https://github.com/jnsgruk/kubernetes-dashboard-operator/blob/main/CONTRIBUTING.md)).

File: metadata.yaml [hunk 2]
```diff
@@ -22,40 +22,38 @@ issues: https://github.com/canonical/hardware-observer-operator/issues
...
+      Download it from https://www.dell.com/support/home/en-us/drivers/driversdetails?driverid=tdghn .
     filename: perccli.deb
 
   sas2ircu-bin:
     type: file
     description: |
       (Optional) un-republish tool.
-      SASIRCU binary file publish by Broadcom.
+      SASIRCU binary file published by Broadcom.
       Download it from https://docs.broadcom.com/docs/12351730 .
```
Comment (Pjack) [inline metadata.yaml:46]: https://docs.broadcom.com/docs/12351735 is better

File: metadata.yaml [hunk 3]
```diff
@@ -22,40 +22,38 @@ issues: https://github.com/canonical/hardware-observer-operator/issues
...
     filename: sas2ircu
 
   sas3ircu-bin:
     type: file
     description: |
       (Optional) un-republish tool.
-      SASIRCU binary file publish by Broadcom.
+      SASIRCU binary file published by Broadcom.
       Download it from https://docs.broadcom.com/docs/SAS3IRCU_P15.zip .
```
Comment (Pjack) [inline metadata.yaml:54]: https://docs.broadcom.com/docs/SAS3IRCU_P16.zip
it's newer

**Comment by dashmage** [comment]
> We also need to update the link in metadata.yaml https://github.com/canonical/hardware-observer-operator/blob/master/metadata.yaml

The link has already been updated in the `metadata.yaml` file.

**Review by Pjack — CHANGES_REQUESTED** [review]

**Review by Pjack — APPROVED** [review]

**Review by jneo8 — APPROVED** [review]

**Pull request merged**